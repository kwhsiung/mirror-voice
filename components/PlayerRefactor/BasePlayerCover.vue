<template>
  <nuxt-link
    class="cover"
    :to="to"
  >
    <img
      v-show="showImg"
      class="cover__img"
      :src="imgSrc"
      alt=""
    >
  </nuxt-link>
</template>

<script>
export default {
  props: {
    to: {
      type: String,
      default: '/'
    },
    imgSrc: {
      type: String,
      default: ''
    }
  },
  computed: {
    showImg() {
      return this.imgSrc && this.imgSrc !== ''
    }
  }
}
</script>

<style lang="stylus" scoped>
.cover
  display inline-block
  d = 34px
  width d
  height d
  &__img
    width 100%
    height 100%
    object-fit cover
</style>
