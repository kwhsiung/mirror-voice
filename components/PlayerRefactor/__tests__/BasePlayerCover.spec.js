import { shallowMount, RouterLinkStub } from '@vue/test-utils'
import BasePlayerCover from '../BasePlayerCover.vue'

describe('BasePlayerCover', () => {
  test('render NuxtLink with "to" received from "to" props', () => {
    const to = '/test/path'
    const wrapper = shallowMount(BasePlayerCover, {
      propsData: {
        to
      },
      stubs: {
        NuxtLink: RouterLinkStub
      }
    })
    const link = wrapper.find(RouterLinkStub)

    expect(link.props().to).toBe(to)
  })

  test('render img with "src" receive from "imgSrc" props', () => {
    const imgSrc = 'https://test/img/url.png'
    const wrapper = shallowMount(BasePlayerCover, {
      propsData: {
        imgSrc
      },
      stubs: {
        NuxtLink: RouterLinkStub
      }
    })
    const img = wrapper.find('img')

    expect(img.attributes().src).toBe(imgSrc)
  })

  test('hide img if imgSrc is missing', () => {
    const wrapper = shallowMount(BasePlayerCover, {
      stubs: {
        NuxtLink: RouterLinkStub
      }
    })
    const img = wrapper.find('img')

    expect(img.element.style.display).toBe('none')
  })
})

describe('snapshot tests', () => {
  test('renders component correctly if we provide efficient props', () => {
    const wrapper = shallowMount(BasePlayerCover, {
      propsData: {
        to: '/test/path',
        imgSrc: 'https://test/img/url.png'
      },
      stubs: {
        NuxtLink: RouterLinkStub
      }
    })

    expect(wrapper.element).toMatchSnapshot()
  })
})
