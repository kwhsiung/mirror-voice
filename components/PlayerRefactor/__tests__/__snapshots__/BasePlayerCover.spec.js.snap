// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasePlayerCover renders component correctly if we provide efficient props 1`] = `
<a
  class="cover"
>
  <img
    alt=""
    class="cover__img"
    src="https://test/img/url.png"
  />
</a>
`;
